File: old_tutorials/01_loading_and_saving_an_image/sample_1/main.cpp
#include <filesystem>
#include <iostream>

#include "main.hpp"

namespace fs = std::filesystem;

int main(int argc, char* argv[]) {

	fs::path executablePath{ fs::path(argv[0]) };

	fs::path inputFile{ (executablePath.parent_path()).append("data/image.png") };

	std::cout << inputFile << std::endl;

	cv::Mat image = cv::imread(inputFile.string(), cv::IMREAD_COLOR);

	if (image.empty()) {

		std::cout << "Error: 'image' is empty" << std::endl;

		return 1;
	}

	cv::imshow("image", image);

	cv::waitKey(0);

	cv::destroyAllWindows();

    return 0;
}

File: old_tutorials/01_loading_and_saving_an_image/sample_1/main.py
import cv2 as cv

from absl import app
from pathlib import Path

def main(argv):

    executable_path = Path(argv[0])

    input_file = (executable_path.parent).joinpath("data/image.png")

    print(input_file)

    image = cv.imread(str(input_file), cv.IMREAD_COLOR)

    if image is None:

        print("Error: 'image' is empty")

        return 1

    cv.imshow("image", image)

    cv.waitKey(0)

    cv.destroyAllWindows()

    return 0


if __name__ == "__main__":

    app.run(main)